src/bitcoind.cpp

```diff
@@ -28,6 +29,79 @@
 const std::function<std::string(const char*)> G_TRANSLATION_FUN = nullptr;
 UrlDecodeFn* const URL_DECODE = urlDecode;
 
+#if HAVE_DECL_FORK
+
+/** Custom implementation of daemon(). This implements the same order of operations as glibc.
+ * Opens a pipe to the child process to be able to wait for an event to occur.
+ *
+ * @returns 0 if succesful, and in child process.
+ *          >0 if succesful, and in parent process.
+ *          -1 in case of error (in parent process).
+ *
+ *          In case of success, endpoint will be one end of a pipe from the child to parent process,
+ *          which can be used with TokenWrite (in the child) or TokenRead (in the parent).
+ */
+int fork_daemon(bool nochdir, bool noclose, TokenPipeEnd &endpoint)
+{
+    // communication pipe with child process
+    std::optional<TokenPipe> umblical = TokenPipe::create();
+    if (!umblical) {
+        return -1; // pipe or pip2 failed.
+    }
+
+    int pid = fork();
+    if (pid < 0) {
+        return -1; // fork failed.
+    }
+    if (pid != 0) {
+        // Parent process gets read end, closes write end.
+        endpoint = umblical->ReadEnd();
```

Comment:
I didn't know the word before (nice name for a parent-child-pipe 😄), but according to the dictionary the spelling is slightly different:
```suggestion
        endpoint = umbilical->ReadEnd();
```